### PR TITLE
Real merkle proofs

### PR DESCRIPTION
**Pull request by musnit**
- Fix up merkle proof generation and proving to match latest real proofs being generated by the relay chain.
 - Enable real proof verification and validator set updates on the validator registry

working :D - e2e tests all pass

coming up in next PR, enabling parachain merkle proofs on the parachain verification and enabling the full e2e verification for messages

### CODE REVIEW

**Comment by musnit** [comment]
@vgeddes @philipstanislaus update: e2e tests all pass. ready for review :)

**Review by philipstanislaus — APPROVED** [review]
LGTM

File: ethereum/contracts/BeefyLightClient.sol [hunk 1]
```diff
@@ -572,21 +569,24 @@ contract BeefyLightClient {
             );
     }
 
+    // To scale encode the byte array, we need to prefix it
+    // with it's length. This is the expected current length of a leaf.
```
[the remark this answers is not in the document]
Comment (musnit) [reply ethereum/contracts/BeefyLightClient.sol:573]: fixed dbff5df

**Comment by musnit** [comment]
included in https://github.com/Snowfork/snowbridge/pull/481

**Pull request closed without merging**